File: Caronae/Ride/RiderCell.m
#import "RiderCell.h"
#import "Caronae-Swift.h"
#import "UIImageView+crn_setImageWithURL.h"

@interface RiderCell ()

@property (weak, nonatomic) IBOutlet UIImageView *photo;
@property (weak, nonatomic) IBOutlet UILabel *nameLabel;
@property (nonatomic, readwrite) User *user;

@end


@implementation RiderCell

- (void)configureWithUser:(User *)user {
    self.user = user;
    self.nameLabel.text = user.firstName;
    
    if (user.profilePictureURL.length > 0) {
        [self.photo crn_setImageWithURL:[NSURL URLWithString:user.profilePictureURL]];
    }
}

@end

File: Caronae/Ride/RiderCell.swift
import UIKit

class RiderCell: UICollectionViewCell {
    
    @IBOutlet weak var photo: UIImageView!
    @IBOutlet weak var nameLabel: UILabel!
    var user: User!
    
    func configure(with user: User) {
        self.user = user
        if nameLabel != nil {
            nameLabel.text = user.firstName
        }
        if let profilePictureURL = user.profilePictureURL, !profilePictureURL.isEmpty {
            photo.crn_setImage(with: URL(string: profilePictureURL))
        }
    }
}
